src/devtools/DevToolsViewProvider.ts

```diff
@@ -18,16 +48,40 @@ export class DevToolsViewProvider implements vscode.WebviewViewProvider {
       panel.webview,
       this._extensionUri,
     );
-
-    panel.webview.onDidReceiveMessage((data) => {
-      devtoolsEvents.emit("fromDevTools", data);
-    });
+    const panelDisposables: vscode.Disposable[] = [];
+    panel.webview.onDidReceiveMessage(
+      (data) => {
+        if (data.source === "apollo-client-devtools") {
+          devtoolsEvents.emit("fromDevTools", data);
+        }
+        if (data.source === "vscode-panel") {
+          if (data.type === "mounted") {
+            sendToDevTools({
+              type: "initializePanel",
+              initialContext: {
+                port:
+                  serverState?.port ||
+                  vscode.workspace
+                    .getConfiguration("apollographql")
+                    .get("devTools.serverPort", 0),
+                listening: !!serverState?.port,
+              },
+            });
+          }
+        }
+      },
```

Comment:
Where previously the panel kickstarted itself, it now notifies the extension, and the extension dispatches the `initializePanel` message with some additional context.